plotly/tests/test_core/test_graph_objs/test_frames.py

```diff
@@ -56,20 +58,37 @@ def test_deeply_nested_layout_attributes(self):
         )
 
     def test_deeply_nested_data_attributes(self):
-        frames = Frames()
-        frames.append({})
-        frames[0].data = [Bar()]
-        frames[0].data[0].marker.color = 'red'
+        #frames = Frames()
+        #frames.append({})
+        #frames[0].data = [Bar()]
+        #frames[0].data[0].marker.color = 'red'
+
+        frames = Frame
+        frames.data = [Bar()]
+        frames.data[0].marker.color = 'red'
+
+        # parse out valid attrs from ._prop_descriptions
+        prop_descrip = frames.data[0].marker.line._prop_descriptions
+        prop_descrip
+
+        raw_matches = re.findall(
+            "\n        [a-z]+|        [a-z]+\n", prop_descrip
+        )
+        matches = []
+        for r in raw_matches:
+            r = r.replace(' ', '')
+            r = r.replace('\n', '')
+            matches.append(r)
 
         # It's OK if this needs to change, but we should check *something*.
         self.assertEqual(
-            frames[0].data[0].marker.line._get_valid_attributes(),
+            set(matches),
```

Comment:
Instead of a regex, a better way to do this might be to assert on  
```
all_attributes = dir(frames[0].data[0].marker.line)
public_attributes = [d for d in all_attributes if d[0] != '_' and d[:1] != 'on']
assertEqual(set(public_attributes), {'cliponaxis',
 'connectgaps',
 'customdata',
 'customdatasrc',
 'dx',
 'dy',
 'error_x',
 'error_y',
 'figure',
 'fill',
 'fillcolor',
 'hoverinfo',
 'hoverinfosrc',
 'hoverlabel',
 'hoveron',
 'hovertext',
 'hovertextsrc',
 'ids',
 'idssrc',
 'legendgroup',
 'line',
 'marker',
 'mode',
 'name',
 'opacity',
 'parent',
 'plotly_name',
 'r',
 'rsrc',
 'selected',
 'selectedpoints',
 'showlegend',
 'stream',
 't',
 'text',
 'textfont',
 'textposition',
 'textpositionsrc',
 'textsrc',
 'to_plotly_json',
 'tsrc',
 'type',
 'uid',
 'unselected',
 'update',
 'visible',
 'x',
 'x0',
 'xaxis',
 'xcalendar',
 'xsrc',
 'y',
 'y0',
 'yaxis',
 'ycalendar',
 'ysrc'})
```